server/sonar-web/src/main/js/apps/system/components/App.tsx

```diff
@@ -18,30 +18,52 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 import * as React from 'react';
+import * as PropTypes from 'prop-types';
 import Helmet from 'react-helmet';
 import ClusterSysInfos from './ClusterSysInfos';
 import PageHeader from './PageHeader';
 import StandAloneSysInfos from './StandAloneSysInfos';
 import throwGlobalError from '../../../app/utils/throwGlobalError';
 import { translate } from '../../../helpers/l10n';
 import { getSystemInfo } from '../../../api/system';
-import { SysInfo } from '../types';
+import { parseQuery, serializeQuery } from '../utils';
+import { SysInfo, Query } from '../types';
+import { RawQuery } from '../../../helpers/query';
 import '../styles.css';
 
+interface Props {
+  location: { pathname: string; query: RawQuery };
+}
+
 interface State {
   loading: boolean;
+  query: Query;
   sysInfoData?: SysInfo;
 }
 
-export default class App extends React.PureComponent<any, State> {
+export default class App extends React.PureComponent<Props, State> {
   mounted: boolean;
-  state: State = { loading: true };
+
+  static contextTypes = {
+    router: PropTypes.object
+  };
+
+  constructor(props: Props) {
+    super(props);
+    this.state = { loading: true, query: parseQuery(props.location.query) };
```

Comment:
I think it's not expensive to parse query each time it is used, so you can avoid putting it in `state`, and `componentWillReceiveProps` can be removed